File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        if(s.length() == 0) return 0;

        int open_bracket = 0, close_bracket = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '('){
                open_bracket++;
            }else{
                if(open_bracket > 0){
                open_bracket--;
                }else{
                    close_bracket++;
                }
            }
        }
        return open_bracket + close_bracket;
    }
}

File: 0921-minimum-add-to-make-parentheses-valid/0921-minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int n = s.length();

        if(n == 0) return 0;
        int open = 0, close = 0;

        for(int i = 0; i < n; i++){
            if(s[i] == '('){
                open++;
            }else if(s[i] == ')' && open > 0){
                open--;
            }else{
                close++;
            }
        }
        return open + close;
    }
};